galley/pkg/crd/naming.go

```diff
@@ -0,0 +1,37 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package crd
+
+import "strings"
+
+// GroupOf returns the APIGroup portion of a fully qualified CRD name.
```

Comment:
we probably should use k8s parser instead of our own? 

https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#ParseGroupKind

Reply:
Neat. Thanks